Update description of "Use the first image type..." option setting.

rdswift opened the pull request:
<!--
    Thank-you for submitting a pull request. Your effort and input is appreciated.

    Please use this template to help us review your change. Not everything is
    required for every change, so please feel free to omit any sections that
    are not relevant to your change.

    Also, please ensure that you've reviewed the guidelines in CONTRIBUTING.md.
-->

### Summary

This is a…

- [x] Correction
- [ ] Addition
- [ ] Restructuring
- [ ] Minor / simple change (like a typo)
- [ ] Other

### Reason for the Change

Resolves #85 

### Description of the Change

Update description for the option.

### Additional Action Required

None.

rdswift commented:
@phw 

I think that I'm still a bit confused.  Can you please look closely at the new description, especially the part about what happens if the option is not checked, because I'm really unclear about that.  Also, is the Note from the original description still required?

![image](https://user-images.githubusercontent.com/16990904/109577668-38772880-7ab3-11eb-8f79-6a775ec09f09.png)

phw commented:
Thanks a lot for tackling this so quickly. This is tricky, and reviewing it I see again why I don't like this option.

So what it does is:

- If this is checked **AND** image is not of type "front" then use the first type of the image as filename
- Otherwise use the name as configured in Cover Art > "Use the following name for images

So this only has an effect on non-front images. The behavior is mostly equivalent to having this script configured for the cover file name:

    $if($eq(%coverart_maintype%,front),cover,%coverart_maintype%)

This is totally stupid to describe in an understandable way :( My attempt:

> This setting changes how Picard names image files other then front images.
> When checked, Picard will use the type of the image (e.g.: back, booklet, etc.) as the filename when saving, as long as
> the type is not front.  If the image has been assigned multiple types, then the first type will be used.  For example,
> if the image is of types "back" and "raw", then "back" will be used for the filename.  If unchecked or if the image is
> of type "front", Picard will use the file name specified in the "Use the following file name for images" setting in the
> :doc:`options_location` section of the :doc:`options_cover` settings.

There are multiple things wrong with this option:

- It applies for all cover art, not only CAA, even if that is implied
- The behavior is confusing. Without a lengthy description it is not clear at all how this interacts with the image naming string
- It is also not clear that this only applies to images that are not "front" images
- It can be completely replicated with a script like above, which again makes this more confusing

The resulting behavior is, however, useful. And only relying on a script is not too great for someone just looking for a simple solution. Maybe we should just move it to the general Cover Art options and give it a separate label or description.

Something like placing it directly after the input for "Use the following file name for images" and label the checkbox something like:

- [ ] For images that are not front images ignore the above file name and always use the primary image type as file name

But that's also a lot of words :(

rdswift commented:
> This is totally stupid to describe in an understandable way :( My attempt:

I like the way you described it.  I'll make the changes.

> There are multiple things wrong with this option:
> 
> * It applies for all cover art, not only CAA, even if that is implied
> * The behavior is confusing. Without a lengthy description it is not clear at all how this interacts with the image naming string
> * It is also not clear that this only applies to images that are not "front" images
> * It can be completely replicated with a script like above, which again makes this more confusing

I see why you're not at all comfortable with it the way it is.

> The resulting behavior is, however, useful. And only relying on a script is not too great for someone just looking for a simple solution. Maybe we should just move it to the general Cover Art options and give it a separate label or description.
> 
> Something like placing it directly after the input for "Use the following file name for images" and label the checkbox something like:
> 
> * [ ]  For images that are not front images ignore the above file name and always use the primary image type as file name
> 
> But that's also a lot of words :(

I like the idea.  Perhaps a shortened text for the option might be:

* [ ] Always use the primary image type as the file name for non-front images

phw commented:
I like your suggestion for the wording. I added https://tickets.metabrainz.org/browse/PICARD-2130 for this change. Let's see to get this for the 2.6 final release.

rdswift commented:
> Let's see to get this for the 2.6 final release.

In that case, I'll hold off merging this PR.  When the changes are made in Picard (and I can grab some screen shots), then I'll make the updates to both the revised settings screens and update the screen shots.

Do you think the next version will have many more changes that require documentation updates?  If so, it might make more sense to create a new `next_2.6_release` branch here to collect the changes, and then just merge that into `master` when the next Picard version is released.

phw commented:
> Do you think the next version will have many more changes that require documentation updates?

The original plan actually was to put this out without bigger changes. The main focus of this release still is fixing those deadlock issues. So I think we won't have much, probably only this configuration restructuring.

phw commented:
I did the refactoring in https://github.com/metabrainz/picard/pull/1757

rdswift commented:
> I did the refactoring in [metabrainz/picard#1757](https://github.com/metabrainz/picard/pull/1757)

I rebased this PR's branch to simplify the merge, and updated it to reflect the refactored option setting screens.  I downloaded the Windows portable artifact from your Picard PR to get the new screen shots.  This should now be ready to merge when the updated version of Picard is released.

phw commented on config/options_location.rst:
```diff
@@ -39,3 +39,17 @@
+
+**Save only a single front image as separate file**
+
+   This tells Picard to only save the first "front" image to a separate file with the release.  If left unchecked, all "front"
```
Maybe something like "... only the first front image and no images of other type...", because this strictly limits to a single image to be saved

rdswift replied:
I changed the wording a bit from your suggestion, and added a sentence to further clarify and emphasize.  Does this properly capture the intent?

Also changed the title as suggested.  We'll have to remember to change the image once your change to Picard has been applied, so I can grab a new screen shot.

phw reviewed (changes requested):
Looks very good to me, only a small suggested change for clarify the single cover file use.

Also could you rename "Only embed a front image" to "Only embed a single front image"? I changed the label here to have it consistent for the same "Save..." option.

phw reviewed (approved):
Oh sorry, I gave you the wrong title. The screenshot is already correct, but the new wording is "Embed only a single front image".

This was to make it consistent with "Save only a single front image as separate file". Apart from that let's get this merged
